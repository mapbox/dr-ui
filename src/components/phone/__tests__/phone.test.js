import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './phone-test-cases.js';

describe('phone', () => {
  describe(testCases.iosHori.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      window.matchMedia = jest.fn().mockImplementation((query) => {
        return {
          matches: false,
          media: query,
          onchange: null,
          addListener: jest.fn(),
          removeListener: jest.fn()
        };
      });

      testCase = testCases.iosHori;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.iosVert.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.iosVert;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe(testCases.androidLandscape.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.androidLandscape;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe(testCases.androidVert.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.androidVert;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
