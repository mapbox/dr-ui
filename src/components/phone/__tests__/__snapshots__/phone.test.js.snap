// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`phone Android landscape renders as expected 1`] = `
<div
  className="shadow-darken25 mx-auto border border--gray-dark relative bg-darken75 wmax600 flex-parent"
  style={
    Object {
      "borderRadius": "30px",
      "borderWidth": "12px",
    }
  }
>
  <div
    className="bg-gray-dark absolute h3 round-t none block-mm"
    style={
      Object {
        "left": "180px",
        "top": "-15px",
        "width": "90px",
      }
    }
  />
  <div
    className="bg-gray-dark absolute h3 round-t none block-mm"
    style={
      Object {
        "left": "90px",
        "top": "-15px",
        "width": "45px",
      }
    }
  />
  <div
    className="flex-parent flex-parent--center-cross z1 relative h-auto bg-gray-dark w36 pl12"
    style={
      Object {
        "borderRadius": "0",
      }
    }
  >
    <div
      className="bg-gray-dark flex-child flex-child--grow flex-parent relative"
      style={
        Object {
          "borderRadius": "0",
        }
      }
    >
      <div
        className="bg-lighten10 round-full flex-child h60 w6"
      />
      <div
        className="bg-lighten10 round-full w6 h6 flex-child absolute"
        style={
          Object {
            "bottom": "-12px",
            "left": undefined,
            "order": undefined,
          }
        }
      />
    </div>
  </div>
  <div
    className="relative flex-parent flex-parent--center-cross flex-parent--center-main  hmin120-mm hmin60"
    style={
      Object {
        "borderRadius": undefined,
        "height": "100%",
        "marginLeft": undefined,
        "marginTop": undefined,
        "overflow": "hidden",
        "width": "calc(100% - 36px)",
      }
    }
  >
    <div
      className="loading loading--dark absolute z0"
    />
    <div
      className="relative"
    >
      <img
        alt="example"
        src="/files/ios-horizontal.png"
      />
    </div>
  </div>
  <div
    className="flex-parent bg-gray-dark flex-parent--center-main flex-parent--center-cross w36"
  >
    <div
      className="bg-lighten10 round-full flex-child h60 w6"
    />
  </div>
</div>
`;

exports[`phone Android portrait renders as expected 1`] = `
<div
  className="shadow-darken25 mx-auto border border--gray-dark relative bg-darken75 wmax300"
  style={
    Object {
      "borderRadius": "30px",
      "borderWidth": "12px",
    }
  }
>
  <div
    className="bg-gray-dark absolute w3 round-r"
    style={
      Object {
        "height": "80px",
        "right": "-15px",
        "top": "180px",
      }
    }
  />
  <div
    className="bg-gray-dark absolute w3 round-r"
    style={
      Object {
        "height": "45px",
        "right": "-15px",
        "top": "90px",
      }
    }
  />
  <div
    className="flex-parent flex-parent--center-cross z1 relative"
    style={
      Object {
        "borderRadius": "0",
      }
    }
  >
    <div
      className="bg-gray-dark flex-child flex-child--grow flex-parent relative flex-parent--center-main mx-auto w-full h36 pt12"
      style={
        Object {
          "borderRadius": "0",
        }
      }
    >
      <div
        className="bg-lighten10 round-full flex-child w60 h6"
      />
      <div
        className="bg-lighten10 round-full w6 h6 flex-child relative"
        style={
          Object {
            "bottom": undefined,
            "left": "-12px",
            "order": -1,
          }
        }
      />
    </div>
  </div>
  <div
    className="relative flex-parent flex-parent--center-cross flex-parent--center-main  hmin300"
    style={
      Object {
        "borderRadius": undefined,
        "height": "calc(100% - 72px)",
        "marginLeft": undefined,
        "marginTop": undefined,
        "overflow": "hidden",
        "width": undefined,
      }
    }
  >
    <div
      className="loading loading--dark absolute z0"
    />
    <div
      className="relative"
    >
      <img
        alt="example"
        src="/files/ios-vertical.png"
      />
    </div>
  </div>
  <div
    className="flex-parent bg-gray-dark flex-parent--center-main flex-parent--center-cross h36"
  >
    <div
      className="bg-lighten10 round-full flex-child w60 h6"
    />
  </div>
</div>
`;

exports[`phone iOS landscape renders as expected 1`] = `
<div
  className="shadow-darken25 mx-auto border border--gray-dark relative bg-darken75 wmax600 flex-parent"
  style={
    Object {
      "borderRadius": "30px",
      "borderWidth": "12px",
    }
  }
>
  <div
    className="bg-gray-dark absolute h3 round-t none block-mm"
    style={
      Object {
        "left": "60px",
        "top": "-15px",
        "width": "90px",
      }
    }
  />
  <div
    className="flex-parent absolute flex-parent--space-between-main w60"
    style={
      Object {
        "bottom": "-15px",
        "left": "55px",
      }
    }
  >
    <div
      className="bg-gray-dark h3 round-b w24 none block-mm"
    />
    <div
      className="bg-gray-dark h3 round-b w24 none block-mm"
    />
  </div>
  <div
    className="flex-parent flex-parent--center-cross z1 relative h-auto w18"
    style={
      Object {
        "borderRadius": "18px 18px 0 0 ",
      }
    }
  >
    <div
      className="bg-gray-dark flex-child flex-child--grow flex-parent relative h120 flex-parent--center-main flex-parent--column flex-parent--column-reverse"
      style={
        Object {
          "borderRadius": "0 12px 12px 0",
        }
      }
    >
      <div
        className="absolute  "
        style={
          Object {
            "background": "radial-gradient(circle at right top, transparent 0, transparent 75%, #23374d 75%, #23374d 100%)",
            "height": "10px",
            "marginLeft": false,
            "top": "-10px",
            "width": "10px",
          }
        }
      />
      <div
        className="bg-lighten10 round-full flex-child h60 w6"
      />
      <div
        className="bg-lighten10 round-full w6 h6 flex-child mb6 mt-neg12"
        style={
          Object {
            "bottom": undefined,
            "left": undefined,
            "order": undefined,
          }
        }
      />
      <div
        className="absolute "
        style={
          Object {
            "background": "radial-gradient(circle at bottom right, transparent 0, transparent 75%, #23374d 75%, #23374d 100%)",
            "bottom": "-10px",
            "height": "10px",
            "marginRight": false,
            "width": "10px",
          }
        }
      />
    </div>
  </div>
  <div
    className="relative flex-parent flex-parent--center-cross flex-parent--center-main  hmin120-mm hmin60"
    style={
      Object {
        "borderRadius": "18px",
        "height": "100%",
        "marginLeft": "-18px",
        "marginTop": undefined,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <div
      className="loading loading--dark absolute z0"
    />
    <div
      className="relative"
    >
      <div>
        <video
          autoPlay={true}
          className="block mx-auto"
          loop={true}
          muted={true}
          src="/files/browser-example.mp4"
          title="example"
          type="video/mp4"
          width="100%"
        >
          <p>
            Your browser doesn't support HTML5 video. Here is a
             
            <a
              href="/files/browser-example.mp4"
            >
              link to the video
            </a>
             instead.
          </p>
        </video>
      </div>
    </div>
  </div>
</div>
`;

exports[`phone iOS portrait renders as expected 1`] = `
<div
  className="shadow-darken25 mx-auto border border--gray-dark relative bg-darken75 wmax300"
  style={
    Object {
      "borderRadius": "30px",
      "borderWidth": "12px",
    }
  }
>
  <div
    className="bg-gray-dark absolute w3 round-r"
    style={
      Object {
        "height": "80px",
        "right": "-15px",
        "top": "90px",
      }
    }
  />
  <div
    className="flex-parent absolute flex-parent--space-between-main flex-parent--column h60"
    style={
      Object {
        "left": "-15px",
        "top": "75px",
      }
    }
  >
    <div
      className="bg-gray-dark w3 round-l h24"
    />
    <div
      className="bg-gray-dark w3 round-l h24"
    />
  </div>
  <div
    className="flex-parent flex-parent--center-cross z1 relative round-t-full"
    style={
      Object {
        "borderRadius": "0",
      }
    }
  >
    <div
      className="bg-gray-dark flex-child flex-child--grow flex-parent relative flex-parent--center-main mx-auto h18 wmax120"
      style={
        Object {
          "borderRadius": "0 0 12px 12px",
        }
      }
    >
      <div
        className="absolute   top left"
        style={
          Object {
            "background": "radial-gradient(circle at bottom left, transparent 0, transparent 75%, #23374d 75%, #23374d 100%)",
            "height": "10px",
            "marginLeft": "-10px",
            "top": false,
            "width": "10px",
          }
        }
      />
      <div
        className="bg-lighten10 round-full flex-child w60 h6"
      />
      <div
        className="bg-lighten10 round-full w6 h6 flex-child ml6 mr-neg12"
        style={
          Object {
            "bottom": undefined,
            "left": undefined,
            "order": undefined,
          }
        }
      />
      <div
        className="absolute  top right"
        style={
          Object {
            "background": "radial-gradient(circle at bottom right, transparent 0, transparent 75%, #23374d 75%, #23374d 100%)",
            "bottom": false,
            "height": "10px",
            "marginRight": "-10px",
            "width": "10px",
          }
        }
      />
    </div>
  </div>
  <div
    className="relative flex-parent flex-parent--center-cross flex-parent--center-main  hmin300"
    style={
      Object {
        "borderRadius": "18px",
        "height": "calc(100% - 18px)",
        "marginLeft": undefined,
        "marginTop": "-18px",
        "overflow": "hidden",
        "width": undefined,
      }
    }
  >
    <div
      className="loading loading--dark absolute z0"
    />
    <div
      className="relative"
    >
      <img
        alt="example"
        src="/files/ios-vertical.png"
      />
    </div>
  </div>
</div>
`;
